source/adapters/hip/device.cpp

```diff
@@ -785,6 +786,145 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetInfo(ur_device_handle_t hDevice,
+        &tex_pitch_align, hipDeviceAttributeTexturePitchAlignment,
+        hDevice->get()));
+    detail::ur::assertion(tex_pitch_align >= 0);
+    return ReturnValue(static_cast<uint32_t>(tex_pitch_align));
+  }
+  case UR_DEVICE_INFO_MAX_IMAGE_LINEAR_WIDTH_EXP: {
+    // Default values due to non-existent hipamd queries for linear sizes.
+    constexpr uint32_t MaxLinearWidth{1};
+    return ReturnValue(MaxLinearWidth);
+  }
+  case UR_DEVICE_INFO_MAX_IMAGE_LINEAR_HEIGHT_EXP: {
+    // Default values due to non-existent hipamd queries for linear sizes.
+    constexpr uint32_t MaxLinearHeight{1};
+    return ReturnValue(MaxLinearHeight);
+  }
+  case UR_DEVICE_INFO_MAX_IMAGE_LINEAR_PITCH_EXP: {
+    // Default values due to non-existent hipamd queries for linear sizes.
+    constexpr uint32_t MaxLinearPitch{1};
+    return ReturnValue(MaxLinearPitch);
+  }
+  case UR_DEVICE_INFO_MIPMAP_SUPPORT_EXP: {
+    // HIP supports mipmaps.
+    // TODO: DPC++ doesn't implement the required mipmap builtins for SYCL.
+    return ReturnValue(ur_bool_t{false});
+  }
+  case UR_DEVICE_INFO_MIPMAP_ANISOTROPY_SUPPORT_EXP: {
+    // HIP supports anisotropic filtering.
+    // TODO: DPC++ doesn't implement the required mipmap builtins for SYCL.
+    return ReturnValue(ur_bool_t{false});
+  }
+  case UR_DEVICE_INFO_MIPMAP_MAX_ANISOTROPY_EXP: {
+    // HIP has no query for this, but documentation states max value is 16.
+    constexpr float MaxAnisotropy{16.f};
+    return ReturnValue(MaxAnisotropy);
+  }
+  case UR_DEVICE_INFO_MIPMAP_LEVEL_REFERENCE_SUPPORT_EXP: {
+    // HIP supports creation of images from individual mipmap levels.
+    // TODO: DPC++ doesn't implement the required mipmap builtins for SYCL.
+    return ReturnValue(ur_bool_t{false});
+  }
+
+  case UR_DEVICE_INFO_EXTERNAL_MEMORY_IMPORT_SUPPORT_EXP: {
+    // Importing external memory is supported, but there are current
+    // issues in the HIP runtime due to which mapping the external array memory
+    // does not work at the moment. Linear/buffer memory mapping is supported.
+    return ReturnValue(ur_bool_t{false});
+  }
+  case UR_DEVICE_INFO_EXTERNAL_SEMAPHORE_IMPORT_SUPPORT_EXP: {
+    // HIP supports importing external semaphores.
+    // TODO: Importing external semaphores should be supported in the adapter,
+    // but there are still current issues as not all related tests are passing.
+    return ReturnValue(ur_bool_t{false});
+  }
+  case UR_DEVICE_INFO_CUBEMAP_SUPPORT_EXP: {
+    // HIP supports cubemaps.
+    // TODO: DPC++ doesn't implement the required builtins for SYCL.
+    return ReturnValue(ur_bool_t{false});
+  }
+  case UR_DEVICE_INFO_CUBEMAP_SEAMLESS_FILTERING_SUPPORT_EXP: {
+    // HIP supports cubemap seamless filtering.
+    // TODO: DPC++ doesn't implement the required builtins for SYCL.
+    return ReturnValue(ur_bool_t{false});
+  }
+  case UR_DEVICE_INFO_BINDLESS_SAMPLED_IMAGE_FETCH_1D_USM_EXP: {
+    // HIP does support fetching 1D USM sampled image data.
```

Comment:
This may be a personal preference but I prefer `supports` (as in the comment(s) above) to `does support` (below). I initially parsed it as `does not support`.